src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,37 @@
+import { useState } from 'react';
+
+type TLocalStorageValue = string | boolean | number;
+
+interface IUseLocalStorageProps<T extends TLocalStorageValue> {
+  localStorageKey: string;
+  initialValue: T;
+  localStorageValueParser?: (localStorageValue: string) => T;
+}
+
+export const useLocalStorage = <T extends TLocalStorageValue>({
+  localStorageKey,
+  initialValue,
+  localStorageValueParser = defaultLocalStorageValueParser<T>,
+}: IUseLocalStorageProps<T>) => {
+  const [storedValue, setStoredValue] = useState<T>(() => {
+    const originalLocalStorageValue = localStorage.getItem(localStorageKey);
+    return originalLocalStorageValue ? localStorageValueParser(originalLocalStorageValue) : initialValue;
+  });
+
+  const storeToLocalStorage = (value: T) => {
+    const stringifiedValue = `${value}`;
+    if (stringifiedValue) {
+      localStorage.setItem(localStorageKey, stringifiedValue);
+    } else {
+      localStorage.removeItem(localStorageKey);
+    }
+
+    setStoredValue(value);
+  };
+
+  return { storedValue, storeToLocalStorage };
```

Comment:
Just `storeToStorage` would be probably enough, `storeValue` could be also an option, but it's very similar to `storedValue`.